pkg/service/accesssvc/create.go

```diff
@@ -68,6 +69,14 @@ func (s *Service) CreateRequest(ctx context.Context, user *identity.User, in typ
 		RuleVersion:     rule.Version,
 		SelectedWith:    make(map[string]access.Option),
 	}
+
+	//validate the request against the access handler - make sure that access will be able to be provisioned
```

Comment:
Can you please shift this logic to be inside the Granter's `CreateGrant` method, rather than exposed as a separate method?

This current implementation means that in future we could accidentally call `granter.CreateGrant()` without calling `granter.ValidateGrant()` first. If the validation logic sits inside of `granter.CreateGrant()`, we'll always call it.